src/Components/ServiceAndTime.jsx

```diff
@@ -0,0 +1,883 @@
+    }
+  };
+
+  // const handleTextClick = () => {
+  //   if (!isChecked.some((checked) => checked)) {
+  //   }
+  // };
+
+  const handleDoneButtonClick = () => {
+    setShowModal(false);
+    setEditAppointmentTime(false)
+    setEditServciceIndex()
+  };
+
+  // Appointment starts
+  const onServiceChange = (index, key, value) => {
+    setAppointmentInput((prevInput) => ({
+      ...prevInput,
+      services: prevInput.services.map((service, i) =>
+        i === index ? { ...service, [key]: value } : service
+      ),
+    }));
+  };
+
+  // const onDurationUnitChange = (index, value) => {
+  //   setAppointmentInput((prevInput) => ({
+  //     ...prevInput,
+  //     services: prevInput.services.map((service, i) =>
+  //       i === index ? { ...service, durationUnit: value } : service
+  //     ),
+  //   }));
+  // };
+
+  // const handleAddService = () => {
+  //   setAppointmentInput((prevInput) => {
+  //     const { services } = prevInput;
+  //     const maxBookingPerSlot = 5;
+
+  //     if (services.length >= maxBookingPerSlot) {
+  //       setIsValidServiceCount(false);
+  //       setValidNumberofServicesError(t("maximumNumOfService1") + maxBookingPerSlot + t("maximumNumOfService2"));
+  //       return prevInput;
+  //     }
+  //     const newService = {
+  //       name: "",
+  //       duration: "",
+  //       // durationUnit: "minutes",
+  //       slotSameAsAppointment: false,
+  //       metadata: {
+  //         holidays: [],
+  //         openingDates: daysOfWeek.reduce((acc, day) => ({ ...acc, [day]: [initialTimeSlot] }), {}),
+  //         maxBookingPerSlot: 5, // Or you can adjust it according to your logic
+  //       },
+  //     };
+
+  //     return {
+  //       ...prevInput,
+  //       services: [...services, newService],
+  //     };
+  //   });
+  // };
+
+  const handleAddService = () => {
+    setAppointmentInput((prevInput) => {
+      const { services, metadata } = prevInput;
+      const maxBookingPerSlot = 5;
+
+      if (services.length >= maxBookingPerSlot) {
+        setIsValidServiceCount(false);
+        setValidNumberofServicesError(
+          t("maximumNumOfService1") + maxBookingPerSlot + t("maximumNumOfService2")
+        );
+        return prevInput;
+      }
+
+      const newService = {
+        name: "",
+        duration: "",
+        slotSameAsAppointment: false,
+        metadata: {
+          holidays: [],
+          openingDates: metadata.openingDates, // Use the current general opening dates
+          maxBookingPerSlot: 5, // Or you can adjust it according to your logic
+        },
+      };
+
+      return {
+        ...prevInput,
+        services: [...services, newService],
+      };
+    });
+  };
+
+  const handleDeleteService = (indexToDelete) => {
+    setAppointmentInput((prevInput) => ({
+      ...prevInput,
+      services: prevInput.services.filter((_, index) => index !== indexToDelete),
+    }));
+  };
+
+  // const [isCheckedList, setIsCheckedList] = useState(
```

Comment:
All those comments are needed. The comments are the changes for individual time slot addition. Currently we commented because app dosent have changes. But they will be implemented afterwards.